### ISSUE TITLE
Crash

### ISSUE THREAD

**Issue by 720mmHowizter** [Ottermandias/GatherBuddy#35]
Game simply completely freezes and crashes when clicking on a node to teleport and swap gear which right now makes the plugin completely unusable with the queue times.
Not doing anything special.

**Comment by BlueRaccoonTech** [comment]
I've also encountered this same issue (and also currently getting upset at myself for not trying this in the morning, when the queue length is more reasonable for testing stuff like this!)

I'm trying to look through my Dalamud logs for any sort of stacktrace or the like, but unfortunately I don't seem to be finding anything of the sort.

**Comment by Ottermandias** [comment]
I do not have any crashes when clicking on nodes. You'll need to provide more information, I guess.

**Comment by BlueRaccoonTech** [comment]
I can say that I was in Radz-at-Han, the node I clicked on was the timed node for Ghostly Umbral Rock, and I had NoSoliciting, Simple Tweaks, RezPls, and GatherBuddy (of course) installed, updated, and enabled at the time of crash. Pixel Perfect was also installed, but disabled due to not being updated yet.

...Now that I think about it, I recall earlier needing to install a certain plugin to have the class change and teleport functionality in the first place, or perhaps I'm misremembering? Either way, I'll try re-installing it and testing some different configurations and things further tomorrow morning, when the queues are less painful.

**Comment by Ottermandias** [comment]
The dependencies on ChatCoordinates and Teleporter were removed in API4, so no worries about that.

**Comment by BlueRaccoonTech** [comment]
So after the reinstall, I appear to be unable to reproduce the same crash that I had mentioned before, even after installing all the same plugins I had mentioned and trying to keep as many variables as possible the same.

I'm guessing whatever crash is occurring may either be something unexpected in configuration being carried over from an update or some other sort of intermittent problem - the former of which I'm unfortunately no longer able to test since I don't believe I kept backups of my configuration, and the latter being much more difficult to spot reliably, and I was just really unlucky to encounter it when I did.

**Comment by pyoio** [comment]
I've just installed launcher / dalamud / gatherbuddy for the first time and I intermittently experience crashes when clicking on a node to trigger teleport. It's happened 4 times in about 14h of gathering. Do I need to do anything special to enable logging for stacktraces? Keen to try and help diagnose, the plugin is amazing, my gathering productivity has shot up 🙏!

**Comment by lionpants** [comment]
My wife and I both crashed yesterday as well when clicking on a row in the UI to teleport to.

We both have the following checkboxes enabled...
- Gear Change
- Teleport
- Map Marker (Open)
- Print Node Uptimes
- Print Gig-head
- Show Fish Timer
- Show Window Timers

Everyone below is default.

If it happens again, are there logs I can upload from somewhere?

**Comment by FLeXyo** [comment]
I've had a similar issue since ShB. The game just crashes to desktop when using `/gather <itemname>` sometimes, maybe like 1/50 times. I think I even checked logs at some point but couldn't find anything (my log levels might be wrong).

**Comment by Ottermandias** [comment]
Does this still happen since the update that disabled map flags?

**Comment by retep998** [comment]
I just updated to the latest version (2.2.1.4) and after teleporting to a few fish nodes just fine, when I tried teleporting to another fish node my game just suddenly crashed

**Comment by kissorjeyabalan** [comment]
Don't know if you've received a stack trace yet - but here's one I got today:

```
Stack trace:
   at System.Environment.get_StackTrace()
   at Dalamud.EntryPoint.VehCallback(IntPtr dumpPath, IntPtr logPath, IntPtr log) in C:\goatsoft\companysecrets\dalamud\EntryPoint.cs:line 68
   at Dalamud.Game.Framework.HandleFrameworkUpdate(IntPtr framework) in C:\goatsoft\companysecrets\dalamud\Game\Framework.cs:line 269

Unhandled native exception occurred at ffxiv_dx11.exe+4D3A2B	(sub_1404D3650+0x3DB)
Code: C0000005
Dump at: C:\Users\kij\AppData\Roaming\XIVLauncher\addon\Hooks\694ac7c\dalamud_appcrash.dmp
Time: 2021-12-28 11:55:26.1412625 CET

Exception Info #1
Address: C0000005
Flags: 0
Address: 7FF6E8BE3A2B
Parameters: 0, 0
Call Stack
{
  [0]	ffxiv_dx11.exe+4D3A2B	(sub_1404D3650+0x3DB)
  [1]	ffxiv_dx11.exe+534D78	(Component::GUI::AtkTextNodeRenderer.Draw+0x88)
  [2]	ffxiv_dx11.exe+536EF3	(Component::GUI::AtkResourceRendererManager.DrawUldFromData+0xA3)
}

Registers
{
  RAX:	2E7000E38A0 [25D39323A31315B]
  RBX:	1FFF
  RCX:	2E7000E5890
  RDX:	1FF077E0
  R8:	F
  R9:	0
  R10:	2E71FFEB080 [25D39323A31315B]
  R11:	2E7000E38A0 [25D39323A31315B]
  R12:	0
  R13:	EAAE55ED90 [4434C00042180000]
  R14:	2E7000E38A0 [25D39323A31315B]
  R15:	0
  RSI:	2E77F1A5418 [2E7000E26B0]
  RDI:	0
  RBP:	2E71FFEB080 [25D39323A31315B]
  RSP:	EAAE55EC30 [1A]
  RIP:	ffxiv_dx11.exe+4D3A2B	(sub_1404D3650+0x3DB)
}

Stack
{
  [RSP+0]	1A
  [RSP+8]	2E71FFEB080 [25D39323A31315B]
  [RSP+10]	2E77F1A5418 [2E7000E26B0]
  [RSP+18]	5000300000004
  [RSP+20]	448E000041600000
  [RSP+28]	5000500000007
  [RSP+30]	448E000042080000
  [RSP+38]	E000400050000
  [RSP+40]	448E000044134000
  [RSP+48]	E000300050004
  [RSP+50]	448E000044180100
  [RSP+58]	FFFF
  [RSP+60]	1A00000000
  [RSP+68]	500040000001B
  [RSP+70]	0
  [RSP+78]	2E7000E38A0 [25D39323A31315B]
}
```

**Comment by Ottermandias** [comment]
Oh, with the named functions that's actually helpful. Thank you very much!

**Comment by cimba007** [comment]
Also had a crash while BTN/MIN. How can I provide more data on the actual crash?

**Comment by retep998** [comment]
I had a crash while selecting a node on MIN/BTN. I had teleports disabled to make sure that wasn't the issue, only changing class + placing marker, and it still crashed (after many instances of selecting a node fine with no crash).

**Comment by Aaron2550** [comment]
I had two crashes so far, they both happened when i typed `/gathermin [some ore]`. Windows Event Viewer doesnt show me any useful Info. I'll attach some stuff that seems atleast somewhat useful...

ExceptionCode: c0000005
ExceptionOffset: 00000000001ca037
ExceptionTimestamp: 607f4c79
Module Version: 5.0.621.22011
Module Name: coreclr.dll
ApplicationTimestamp: 61ba4fdc

Google told me the Exception Code `c0000005` is for a Memory Access Violation but thats all i could find since Windows doesnt log anymore than that...

**Comment by Ottermandias** [comment]
Hopefully fixed due to using macro system in 3.0
